psi4/src/psi4/libscf_solver/hf.cc

```diff
@@ -1450,5 +1458,92 @@ bool HF::stability_analysis() {
     throw PSIEXCEPTION("Stability analysis hasn't been implemented yet for this wfn type.");
     return false;
 }
+
+extern "C" const int64_t obj_func_wrapper(const double* kappa, double* func) {
+    if (!HF::instance) throw PSIEXCEPTION("No HF instance set!\n");
+    return HF::instance->obj_func(kappa, func);
+}
+
+extern "C" int64_t hess_x_wrapper(const double* x, double** hess_x) {
+    if (!HF::instance) throw PSIEXCEPTION("No HF instance set!\n");
+    return HF::instance->hess_x(x, hess_x);
+}
+
+extern "C" const int64_t update_orbs_wrapper(const double* kappa, double* func, double** grad,
+    double** h_diag, int64_t (**hess_x_out)(const double*, double**)) {
+    if (!HF::instance) throw PSIEXCEPTION("No HF instance set!\n");
+    return HF::instance->update_orbs(kappa, func, grad, h_diag, hess_x_out);
+}
+
+extern "C" const void logger(const char* message) {
+    outfile->Printf(" %s\n", message);
+}
+
+void HF::opentrustregion_scf() {
+    // check if OpenTrustRegion is enabled and loading of dynamic libraries is possible
+    #ifndef USING_OpenTrustRegion
+        throw PSIEXCEPTION("OpenTrustRegion support has not been enabled in this Psi4 build!\n");
+    #elifndef HAVE_DLFCN_H
+        throw PSIEXCEPTION("Plugins are not supported on your platform.\n");
+    #endif
+
+    // load library
+    void* handle = dlopen("libopentrustregion.dylib", RTLD_LAZY);
```

Comment:
this looks like a macosx hack, no?

I think the library should be linked in dynamically